### PR TITLE
Fixed with added rules configurator and rule files

### PR DESCRIPTION
**Pull request by freschri**
Note: some changes in here are also in https://github.com/aws-observability/cdk-aws-observability-accelerator/pull/76

### CODE REVIEW

File: lib/single-new-eks-opensource-observability-construct/amp-rules-configurator-addon.ts
Comment (elamaran11) [inline lib/single-new-eks-opensource-observability-construct/amp-rules-configurator-addon.ts:39]: `@dependable` is required for dependent addons like AmpAddon etc. If not this might run ahead of time.

File: lib/single-new-eks-opensource-observability-construct/rules-1.ytpl
Comment (elamaran11) [inline lib/single-new-eks-opensource-observability-construct/rules-1.ytpl:1]: These are lengthy Yamls. Do we have an opprtunity to use GitOps to deploy this to EKS cluster vs deploying it via cluster addon. We have a problem of maintenance in static and with GitOps its better way of managing kubernetes resoruces?

**Review by elamaran11 — CHANGES_REQUESTED** [review]
@freschri This is awesome work. I forgot to add this issue or work on this issue. I appreciate you doing this. Only feedback is we should move this to GitOps pattern via flux than doing via static yamls in code.

**Comment by elamaran11** [comment]
@freschri This is AWSome work, I tried to run this solution and im getting following error. I think you have missed on the dependencies to deploy this solution (which i believe you might have deployed manually while developing). Please check with terraform solution for parity and if you need `prometheusservice.services.k8s.aws` our ACK addon in blueprints supports it. Also Aleriting is missing as you see in Terraform repo.

```
10:18:36 AM | CREATE_FAILED        | Custom::AWSCDK-EKS-KubernetesResource | AmpRulesConfigurat...0/Resource/Default
Received response status [FAILED] from custom resource. Message returned: Error: b'error: resource mapping not found for name: "rule-1" namespace: "ack-system" from "/tmp/manifest.y
aml": no matches for kind "RuleGroupsNamespace" in version "prometheusservice.services.k8s.aws/v1alpha1"\nensure CRDs are installed first\n'
```
In order to move forward with following next steps :
1. Fix all issues in this PR.
2. Apply this solution to other open source patterns like existing cluster and graviton open source accelerator. Also make sure you test it
3. Doc needs to be updated with metrics from other boards which is flowing with this change. 
4. Once you have it, we can test and merge it but this will be a temporary solution for few weeks until we have flux multi path which you are working on.
5. Once we have Flux multi path support in Blueprints next release, you can work making a permanent change to use GitOps for alerts and rules for all 3 patterns and submit new PR.

File: cdk.json
Comment (elamaran11) [inline cdk.json:26]: This cant be removed, This is required for other patterns.

**Review by elamaran11 — CHANGES_REQUESTED** [review]
@freschri Nice work on this. Couple of comments and we are almost there. Also please checkin the documentation changes as well along with showing screenshots of remaining boards working after this change.

File: lib/common/addons/amp-rules-configurator/amp-rules-configurator-addon.ts
Comment (shapirov103) [inline lib/common/addons/amp-rules-configurator/amp-rules-configurator-addon.ts:25]: If it is an extensions addon, then docs are needed.  Also, don't you require a dependency on the AmpAddOn here? Please confirm.

**Review by shapirov103 — COMMENTED** [review]
Please see my comment

**Comment by elamaran11** [comment]
@freschri FYI, i tested this PR and everything works fine, i also see that i can see data in all dashboards. So once you fix the doc, we can review and merge it. Nice work!

File: cdk.json
Comment (freschri) [reply cdk.json:26]: done

File: lib/common/addons/amp-rules-configurator/amp-rules-configurator-addon.ts
Comment (freschri) [reply lib/common/addons/amp-rules-configurator/amp-rules-configurator-addon.ts:25]: @shapirov103 @elamaran11 thanks for your comments, this one is common from both of you. Answering here:
actually the only dependency for this addon is the workspace in AMP, which is provided by:
'''
.resourceProvider(ampWorkspaceName, new blueprints.CreateAmpProvider(ampWorkspaceName, ampWorkspaceName))
'''

the AMP addon deploys the OpenTelemetryCollector for AMP exporting and related permissions. (Maybe a better name would be OpenTelemetryCollectorForAMPAddOn?) so I believe there is no dependency: you can AmpRulesConfiguratorAddOn without AMP addon and it still works because it only interacts with AMP, not the OpenTelemetryCollector. Anything I am missing?

File: docs/addon-extensions/amp-rules-configurator-addon.md
Comment (elamaran11) [inline docs/addon-extensions/amp-rules-configurator-addon.md:1]: Is this doc page obselete if you are dependent on the framework for this change?

**Review by elamaran11 — CHANGES_REQUESTED** [review]
@freschri Nice progress, we are almost close and here is my feedback :
1. If you are making change to existing pattern, the documentation of existing pattern should also be updated with all changes and screenshots.
2. why is the addon-exstension doc for amp-rules required here if the amp-rules are moving to the framework cdk blueprints.
3. Are you going to wait on this PR until blueprints release happens or you want to create an interim AmpRules Addon class and then retrofit the Amp Addon changes later? I think we should follow a two stage approach without waiting for the release.
4. Also fix the GH action errors

**Comment by elamaran11** [comment]
@freschri Do we have an update on this PR? This is an important feature which we should move forward to close soon. Also we need to close the blueprints AmpAddon PR which has this dependency

File: docs/addon-extensions/amp-rules-configurator-addon.md
Comment (freschri) [reply docs/addon-extensions/amp-rules-configurator-addon.md:1]: correct, removed

File: lib/single-new-eks-opensource-observability-construct/graviton-index.ts
Comment (freschri) [reply lib/single-new-eks-opensource-observability-construct/graviton-index.ts:42]: correct, dependent on quickstart release

**Comment by freschri** [comment]
> @freschri Do we have an update on this PR? This is an important feature which we should move forward to close soon. Also we need to close the blueprints AmpAddon PR which has this dependency

done updates; PR dependent on quickstart release

**Review by elamaran11 — CHANGES_REQUESTED** [review]
@freschri Looks good to me but you are missing to update existing-cluster OSS pattern for similar screenshots. Please update that. Also make sure when you pull from upstream

**Comment by freschri** [comment]
> @freschri Looks good to me but you are missing to update existing-cluster OSS pattern for similar screenshots. Please update that. Also make sure when you pull from upstream

I guess no updates? existing oss says on main branch:
Please see [Single New EKS Open Source Observability Accelerator](../single-new-eks-observability-accelerators/single-new-eks-opensource-observability.md).

**Review by elamaran11 — APPROVED** [review]
LGTM

**Comment by elamaran11** [comment]
> > @freschri Looks good to me but you are missing to update existing-cluster OSS pattern for similar screenshots. Please update that. Also make sure when you pull from upstream
> 
> I guess no updates? existing oss says on main branch: Please see [Single New EKS Open Source Observability Accelerator](../single-new-eks-observability-accelerators/single-new-eks-opensource-observability.md).

Makes sense. LGTM.

**Pull request merged**
